### DISCUSSION TITLE
💥 handle server actions errors inside server components 🤔

### DISCUSSION THREAD

**Discussion by aymanechaaba1** [vercel/next.js#60259] [App Router]
Is there a way to handle errors inside server components, (no hooks needed). 🤔

**Comment by ananuness** [comment]
Hi! Actually you have three ways to handle Server actions errors in Server components:
- You can use the `try/catch` block to catch errors and handle them in the catch block.
- You can use a React `ErrorBoundary` component to catch errors
- You can also use the special [`error.tsx`](https://nextjs.org/docs/app/building-your-application/routing/error-handling#handling-server-errors) file to handle errors at layout level.

[There](https://makerkit.dev/how-to/next-supabase/server-actions-error-handling) is some examples about this topic. I hope it helps, but don't hesitate to ask if there's any question!
